cloud: add Netlify integration (#11393)
pingcap/docs: tidb-cloud/integrate-tidbcloud-with-netlify.md

### A Netlify account and a deployed site

You are expected to have an account and a site in Netlify. If you do not have any, refer to the following links to create one:

* [Sign up a new account](https://app.netlify.com/signup).
* [Add a site](https://docs.netlify.com/welcome/add-new-site/) in Netlify. If you do not have an application to deploy, you can use the [TiDB Cloud Starter Template](https://github.com/tidbcloud/nextjs-prisma-example#deploy-on-netlify) to have a try.

### A TiDB Cloud account and a TiDB cluster

You are expected to have an account and a cluster in TiDB Cloud. If you do not have any, refer to [Create a TiDB cluster](/tidb-cloud/create-tidb-cluster.md).

One TiDB Cloud cluster can connect to multiple Netlify sites.

### All IP addresses allowed for traffic filter in TiDB Cloud

For Dedicated Tier clusters, make sure that the traffic filter of the cluster allows all IP addresses (set to `0.0.0.0/0`) for connection, this is because Netlify deployments use dynamic IP addresses.

Serverless Tier clusters allow all IP addresses for connection by default, so you do not need to configure any traffic filter.

## Connect via manually setting environment variables

> To use this method, make sure that you have set the **Allow Access from Anywhere** traffic filter in the [**Security Settings**](/tidb-cloud/configure-security-settings.md) dialog and save the password.

Reviewer: 这里要改一下，allow access from anywhere 是 dedicated 的，serverless 不需要。我觉得改成 have set the traffic filter，然后链接到上面的那个章节。

> To use this method, make sure that your TiDB Cloud cluster [allows all IP addresses access](#all-ip-addresses-allowed-for-traffic-filter-in-tidb-cloud) and save your root password in the [Security Settings](/tidb-cloud/configure-security-settings.md) dialog.

Reviewer 2: _root password_ should also be put in the *prerequisites* section. WDYT?

Author: I don't think _root password_  is a part of _prerequisites_. We could consider putting _root password_ in the following steps.

Reviewer 2: Sounds good. Please make it a step. Otherwise, it might be ignored.

1. Follow the steps in [Connect to a TiDB Cloud cluster via standard connection](/tidb-cloud/connect-to-tidb-cluster.md#connect-via-standard-connection) to get the connection information of your TiDB cluster.

> 2. Go to your **Netlify dashboard** > **Netlify project** > **Site settings** > **Environment Variables**, and then [Update variables](https://docs.netlify.com/environment-variables/get-started/#update-variables-with-the-netlify-ui) according to the connection information of your TiDB cluster.

Reviewer 2: ```suggestion
2. Go to your **Netlify dashboard** > **Netlify project** > **Site settings** > **Environment Variables**, and then [update variables](https://docs.netlify.com/environment-variables/get-started/#update-variables-with-the-netlify-ui) according to the connection information of your TiDB cluster.

    - **Key**: `DATABASE_URL` 
    - **Values**: `mysql://<User>:<Password>@<Endpoint>:<Port>/<Database>?sslaccept=strict`
```

> The following is a datasource example in the Prisma schema file for a TiDB Cloud Serverless Tier cluster:
> 
> ```
> datasource db {
>   provider = "mysql"
>   url      = env("DATABASE_URL")
> }
> ```

Reviewer 2: Can we make this a step? like:

````suggestion
3. Configure the datasource example in the Prisma schema file for a TiDB Cloud Serverless Tier cluster as follows:

```
datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}
```
````

Author: This is a Prisma app example. There are many other ways to connect to TiDB in Netlify. So, it is not the third step.

Reviewer 2: Got it. Please reword this for clarity. Users might get confused if they are not familiar with Prisma or JS/TS.

### All IP addresses allowed for traffic filter in TiDB Cloud

For Dedicated Tier clusters, make sure that the traffic filter of the cluster allows all IP addresses (set to `0.0.0.0/0`) for connection, this is because Netlify deployments use dynamic IP addresses.

Serverless Tier clusters allow all IP addresses for connection by default, so you do not need to configure any traffic filter.

## Connect via manually setting environment variables

1. Make sure that your TiDB Cloud cluster [allows all IP addresses access](#all-ip-addresses-allowed-for-traffic-filter-in-tidb-cloud) and save your root password in the [Security Settings](/tidb-cloud/configure-security-settings.md) dialog.
2. Follow the steps in [Connect to a TiDB Cloud cluster via standard connection](/tidb-cloud/connect-to-tidb-cluster.md#connect-via-standard-connection) to get the connection information of your TiDB cluster.
3. Go to your **Netlify dashboard** > **Netlify project** > **Site settings** > **Environment Variables**, and then [update variables](https://docs.netlify.com/environment-variables/get-started/#update-variables-with-the-netlify-ui) according to the connection information of your TiDB cluster.

> Here we use a Prisma application as an example. The following is a datasource setting in the Prisma schema file for a TiDB Cloud Serverless Tier cluster:
> 
> ```
> datasource db {
>   provider = "mysql"
>   url      = env("DATABASE_URL")
> }
> ```
> 
> In Netlify, you can declare the environment variables as follows.
> 
> - **Key** = DATABASE_URL 
> - **Values** = `mysql://<User>:<Password>@<Endpoint>:<Port>/<Database>?sslaccept=strict`

Reviewer 2: ```suggestion
    Here we use a Prisma application as an example. The following is a datasource setting in the Prisma schema file for a TiDB Cloud Serverless Tier cluster:

    ```
    datasource db {
      provider = "MySQL"
      url      = env("DATABASE_URL")
    }
    ```

    In Netlify, you can declare the environment variables as follows.

    - **Key** = DATABASE_URL 
    - **Values** = `mysql://<User>:<Password>@<Endpoint>:<Port>/<Database>?sslaccept=strict`
```

> 

Reviewer 2: ```suggestion

    You can get the information of `<User>`, `<Password>`, `<Endpoint>`, `<Port>`, and `<Database>` in the TiDB Cloud console.

```

![Set an environment variable in Netlify](/media/tidb-cloud/integration-netlify-environment-variables.jpg)

> Then you can use this new environment variable to connect to your TiDB Cloud cluster after re-deploying the site.

Reviewer 2: ```suggestion
After re-deploying the site, you can use this new environment variable to connect to your TiDB Cloud cluster.
```

1. Follow the steps in [Connect to a TiDB Cloud cluster via standard connection](/tidb-cloud/connect-to-tidb-cluster.md#connect-via-standard-connection) to get the connection information of your TiDB cluster.
2. Go to your **Netlify dashboard** > **Netlify project** > **Site settings** > **Environment Variables**, and then [update variables](https://docs.netlify.com/environment-variables/get-started/#update-variables-with-the-netlify-ui) according to the connection information of your TiDB cluster.

    - **Key**: `DATABASE_URL` 
    - **Values**: `mysql://<User>:<Password>@<Endpoint>:<Port>/<Database>?sslaccept=strict`

The following is a datasource example in the Prisma schema file for a TiDB Cloud Serverless Tier cluster:

```
datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}
```

In Netlify, you can declare the environment variables as follows. You can customize the name according to your project need.

- **Key** = DATABASE_URL 

> - **Values** = `mysql://<User>:<Password>@<Endpoint>:<Port>/<Database>?sslaccept=strict`

Reviewer 2: 1. Since this description repeats step 2, can you remove it?
2. "You can customize the name" means you can customize the **Key**?

> ![img](/media/tidb-cloud/integration-netlify-environment-variables.jpg)

Reviewer 2: Just for accessibility:

```suggestion
![Set an environment variable in Netlify](/media/tidb-cloud/integration-netlify-environment-variables.jpg)
```

* [Adding a site](https://docs.netlify.com/welcome/add-new-site/) in Netlify, or if you do not have an application to deploy, you can use the [TiDB Cloud Starter Template](https://github.com/pingcap/tidb-prisma-vercel-demo#deploy-on-netlify) to have a try.

### A TiDB Cloud account and a TiDB cluster

You are expected to have an account and a cluster in TiDB Cloud. If you do not have any, refer to [Create a TiDB cluster](/tidb-cloud/create-tidb-cluster.md).

To integrate with Netlify, you are expected to have the "Owner" access to your organization or the "Member" access to the target project in TiDB Cloud. For more information, see [Configure member roles](/tidb-cloud/manage-user-access.md#configure-member-roles).

One TiDB Cloud cluster can connect to multiple Netlify sites.

### All IP addresses allowed for traffic filter in TiDB Cloud

For Dedicated Tier clusters, make sure that the traffic filter of the cluster allows all IP addresses (set to `0.0.0.0/0`) for connection, this is because Netlify deployments use dynamic IP addresses.

Serverless Tier clusters allow all IP addresses for connection by default, so you do not need to configure any traffic filter.

> ## Connect via manually setting environment variables

Reviewer 2: Will there be any other method?

Author: Now, there is only one way.
